src/main/java/com/snowflake/kafka/connector/internal/FileNameUtils.java

```diff
@@ -78,11 +89,34 @@ static String brokenRecordFileName(String prefix, long offset, boolean isKey) {
    *
    * @param appName connector name
    * @param table table name
+   * @param topic topic name
    * @param partition partition index
    * @return file prefix
    */
-  static String filePrefix(String appName, String table, int partition) {
-    return appName + "/" + table + "/" + partition + "/";
+  static String filePrefix(String appName, String table, String topic, int partition) {
+    if (partition >= 0x8000) {
+      throw new IllegalArgumentException(
+          String.format("partition id=%d is too large (max=%d)", partition, 0x8000));
+    }
+    BigInteger partitionPart = BigInteger.valueOf(partition);
```

Comment:
Nit: We could extract it to a separate class or method

Reply:
yes, and also move the methods used only in tests from production code to test code